File: src/main/java/dev/kaato/pluginscrud/Java/entities/enums/Stage.java
package dev.kaato.pluginscrud.Java.entities.enums;

import java.util.Arrays;

public enum Stage {
    IN_PLANNING(0),
    IN_DEVELOPMENT(1),
    READY(2);

    private final int id;

    Stage(int id) {
        this.id = id;
    }

    public int getId() {
        return id;
    }

    public static Stage valueOf(int stageId) {
        return Arrays.stream(Stage.values()).filter(it -> it.getId() == stageId).findFirst().get();
    }
}

File: src/main/java/dev/kaato/pluginscrud/kotlin/entities/enum/Stage.kt
package dev.kaato.pluginscrud.kotlin.entities.enum

enum class Stage(val id: Int) {
    IN_PLANNING(0),
    IN_DEVELOPMENT(1),
    READY(2);

    companion object {
        fun valueOf(id: Int): Stage? {
            return entries.find { it.id == id }
        }
    }
}